### PR TITLE
Cleanup in config package.

### PR DESCRIPTION
**Pull request by bogdandrutu**
Summary of the PR:

- Remove unimplemented method from TraceConfig.
- Move config implementations to config package.
- Add tests for NoopTraceConfig.

### CODE REVIEW

File: api/src/test/java/io/opencensus/trace/config/TraceConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+package io.opencensus.trace.config;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import io.opencensus.trace.Samplers;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link TraceParams}. */
```
Comment (sebright) [inline api/src/test/java/io/opencensus/trace/config/TraceConfigTest.java]: Should this say `{@link TraceConfig}`?

File: api/src/test/java/io/opencensus/trace/config/TraceConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+public class TraceConfigTest {
+  TraceConfig traceConfig = TraceConfig.getNoopTraceConfig();
+
+  @Test
+  public void defaultActiveTraceParams() {
+    assertThat(traceConfig.getActiveTraceParams()).isEqualTo(TraceParams.DEFAULT);
+  }
+
+  @Test
+  public void updateActiveTraceParams() {
```
Comment (sebright) [inline api/src/test/java/io/opencensus/trace/config/TraceConfigTest.java]: This test's name could be clearer.  It sounds like the update should succeed, but it doesn't with the no-op implementation.

Comment (bogdandrutu) [inline api/src/test/java/io/opencensus/trace/config/TraceConfigTest.java]: Sure. Updated.

File: api/src/test/java/io/opencensus/trace/config/TraceConfigTest.java (hunk 1 above)
Comment (bogdandrutu) [inline api/src/test/java/io/opencensus/trace/config/TraceConfigTest.java]: Done.

**Pull request merged**